app/main.py

```diff
@@ -1,2 +1,81 @@
+import dataclasses
+from typing import Any
+
+
+@dataclasses.dataclass()
+class Node:
+    key: Any
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.size = 0
+        self.threshold = round(self.capacity * 2 / 3)
+        self.hash_table = [None] * self.capacity
+
+    def collision(self, node: Node, node_ind: int) -> int:
+        curr_id = node_ind
+        while True:
+            if curr_id < len(self.hash_table):
```

Comment:
You can move this check directly in the loop

Reply:
But I can't change current_id to 0 if I haven't found free space. Or is there some way?